src/bank.rs

```diff
@@ -795,9 +812,13 @@ mod tests {
     #[test]
     fn test_process_ledger() {
         let (ledger, pubkey) = create_sample_ledger();
+        let (ledger, dup) = ledger.tee();
         let bank = Bank::default();
-        bank.process_ledger(ledger).unwrap();
+        let (ledger_height, tail) = bank.process_ledger(ledger).unwrap();
```

Comment:
Can you add a second `process_ledger` test for when the ledger is longer than the tail? It should probably test that the last `Entry.id` in the window matches `bank.last_id()`.